Update Windows instructions to WSL2 (#117)
sylabs/singularity-admindocs: installation.rst

Linux container runtimes like {Singularity} cannot run natively on
Windows or Mac because of basic incompatibilities with the host kernel.
(Contrary to a popular misconception, macOS does not run on a Linux
kernel. It runs on a kernel called Darwin originally forked from BSD.)

To run {Singularity} on a Windows or macOS computer, a Linux virtual machine
(VM) is required. There are various ways to configure a VM on both Windows and

> macOS. We recommend the Windows Subsystem for Linux (WSL2), and Lima.

Reviewer: ```suggestion
macOS. On Windows, we recommend the Windows Subsystem for Linux (WSL2),
and on macOS, we recommend Lima.
```

Windows
=======

Recent builds of Windows 10, and all builds of Windows 11, include version 2 of
the Windows Subsystem for Linux. WSL2 provides a Linux virtual machine that is
tightly integrated with the Windows environment. The default Linux distribution
used by WSL2 is Ubuntu. It is straightforward to install {Singularity} inside
WSL2 Ubuntu, and use all of its features.

Follow the `WSL2 installation instructions
<https://docs.microsoft.com/en-us/windows/wsl/install>`__ to enable WSL2 with
the default Ubuntu 22.04 environment. On Windows 11 and the most recent builds
of Windows 10 this is as easy as opening an administrator command prompt or
Powershell window and entering:

.. code::

  wsl --install

Follow the prompts. A restart is required, and when you open the 'Ubuntu' app

> for the first time you’ll be asked to set a username and password for the Linux

Reviewer: ```suggestion
for the first time you'll be asked to set a username and password for the Linux
```

Reviewer: (Our linters produce warnings about "smart quotes", I think)

environment.

You can install SingularityCE from source, or from the Ubuntu packages at the
GitHub releases page. To quickly install the 4.0.0 package use the following
commands inside the WSL2 Ubuntu window:

.. code::

>   $ wget https://github.com/sylabs/singularity/releases/download/v4.0.0/singularity-ce_4.0.0-jammy_amd64.deb

Reviewer: This is currently 404 (as it should be, of course, since the release hasn't happened yet), but we should remember to double-check that the link works once the release is out.

   $ limactl delete singularity-ce

{Singularity} Docker Image
==========================

It is also possible to run {Singularity} inside Docker, or another compatible
OCI container runtime. This may be convenient if you have Docker Desktop, or a
similar solution, already installed on your PC or Mac.

Docker containers for {Singularity} are maintained at
https://quay.io/repository/singularity/singularity. 

.. note::

  These containers are maintained by a third party. They are not part of the

>   {Singularity} project, or reviewed by Sylabs.

Reviewer: ```suggestion
  {Singularity} project, nor are they reviewed by Sylabs.
```

An example of a suitable ``compose.yaml`` file to start up {Singularity} in a
Docker container is given below. Note that privileged operation is needed to
successfully run {Singularity} nested inside of Docker. Change the version
number on the ``image:`` line to your preferred release.

.. code::

   services:
     singularity:
       image: quay.io/singularity/singularity:v3.11.4-slim
       stdin_open: true
       tty: true
       privileged: true
       volumes:
         - .:/root
       entrypoint: ["/bin/sh"]

Singularity in Docker can have various disadvantages, but basic
container operations will work. Currently, the intended use case is

> continuous integration, meaning that you should be able to build a

Reviewer: Is [this](https://sylabs.io/2023/08/using-singularity-container-services-registry-for-ci-cd-workflows/) relevant, and is so, should we link to it here?
